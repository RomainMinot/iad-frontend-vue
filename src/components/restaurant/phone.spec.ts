import { mount } from '@vue/test-utils';

import RestaurantPhone from './phone.vue';
import restaurantsResponse from '@/mock/restaurants.json';

const { display_phone } = restaurantsResponse[0];

describe(`restaurant-phone`, () => {
  test(`no props`, (context) => {
    const wrapper = mount(RestaurantPhone, {
      global: { plugins: [context.router] },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it(`renders correctly a phone number`, (context) => {
    const wrapper = mount(RestaurantPhone, {
      global: { plugins: [context.router] },
      props: { displayPhone: display_phone },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it(`renders correctly an empty phone number`, (context) => {
    const wrapper = mount(RestaurantPhone, {
      global: { plugins: [context.router] },
      props: { displayPhone: undefined },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
