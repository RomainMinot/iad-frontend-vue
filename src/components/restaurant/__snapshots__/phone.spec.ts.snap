// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`restaurant-phone > no props 1`] = `
"<div class=\\"flex gap-2\\"><i class=\\"mdi-phone mdi v-icon notranslate v-theme--light v-icon--size-default text-slate-300\\" aria-hidden=\\"true\\"></i>
  <div class=\\"text-slate-600\\">
    <p class=\\"text-body-1\\">Phone number not specified</p>
  </div>
</div>"
`;

exports[`restaurant-phone > renders correctly a phone number 1`] = `
"<div class=\\"flex gap-2\\"><i class=\\"mdi-phone mdi v-icon notranslate v-theme--light v-icon--size-default text-slate-300\\" aria-hidden=\\"true\\"></i>
  <div class=\\"text-slate-600\\">
    <p class=\\"text-body-1\\">+49 40 57283299</p>
  </div>
</div>"
`;

exports[`restaurant-phone > renders correctly an empty phone number 1`] = `
"<div class=\\"flex gap-2\\"><i class=\\"mdi-phone mdi v-icon notranslate v-theme--light v-icon--size-default text-slate-300\\" aria-hidden=\\"true\\"></i>
  <div class=\\"text-slate-600\\">
    <p class=\\"text-body-1\\">Phone number not specified</p>
  </div>
</div>"
`;
